Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
+
+				// Set default checkbox states and enable or disable them.
+				int defaultFogOfWar = selectedGA->GetDefaultFogOfWar();
+				if (defaultFogOfWar > -1) {
+					m_FogOfWarCheckbox->SetCheck(defaultFogOfWar != 0);
+				}
+				m_FogOfWarCheckbox->SetEnabled(selectedGA->GetFogOfWarSwitchEnabled());
+
+				int defaultReqClearPath = selectedGA->GetDefaultRequireClearPathToOrbit();
+				if (defaultReqClearPath > -1) {
+					m_RequireClearPathToOrbitCheckbox->SetCheck(defaultReqClearPath != 0);
+				}
+				m_RequireClearPathToOrbitCheckbox->SetEnabled(selectedGA->GetRequireClearPathToOrbitSwitchEnabled());
+
+				int defaultDeployUnits = selectedGA->GetDefaultDeployUnits();
+				if (defaultDeployUnits > -1) {
+					m_DeployUnitsCheckbox->SetCheck(defaultDeployUnits != 0);
+				}
+				m_DeployUnitsCheckbox->SetEnabled(selectedGA->GetDeployUnitsSwitchEnabled());
 			}
-			m_pGoldSlider->SetEnabled(pSelectedGA->GetGoldSwitchEnabled());
-
-			//Set default fog of war flag and enable or disable it if necessary
-			if (pSelectedGA->GetDefaultFogOfWar() > -1)
-			{
-				if (pSelectedGA->GetDefaultFogOfWar() == 0)
-					m_pFogOfWarCheckbox->SetCheck(0);
-				else
-					m_pFogOfWarCheckbox->SetCheck(1);
+		} else {
+			m_ScenarioScenes = nullptr;
+			m_ActivityLabel->SetText("No Activity selected.");
+			m_DifficultyLabel->SetVisible(false);
+			if (m_DifficultySlider) {
+				m_DifficultySlider->SetVisible(false);
 			}
-			m_pFogOfWarCheckbox->SetEnabled(pSelectedGA->GetFogOfWarSwitchEnabled());
-
-			//Set default clear path to orbit flag and enable or disable it if necessary
-			if (pSelectedGA->GetDefaultRequireClearPathToOrbit() > -1)
-			{
-				if (pSelectedGA->GetDefaultRequireClearPathToOrbit() == 0)
-					m_pRequireClearPathToOrbitCheckbox->SetCheck(0);
-				else
-					m_pRequireClearPathToOrbitCheckbox->SetCheck(1);
+
+			int textHeight = m_ActivityLabel->ResizeHeightToFit();
+			int padding = 125;
+			m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox)->Resize(m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox)->GetWidth(), textHeight + padding);
+		}
+
+		KeepBoxInScreenBounds(m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox));
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioGUI::UpdatePlayersBox() {
+		RTEAssert(m_SelectedActivity && m_SelectedScene, "Trying to start a scenario game without an activity or a scene.");
+
+		int mouseX = 0;
+		int mouseY = 0;
+		m_GUIInput->GetMousePosition(&mouseX, &mouseY);
+		const GameActivity *gameActivity = dynamic_cast<const GameActivity *>(m_SelectedActivity);
+
+		const GUICollectionBox *hoveredCell = dynamic_cast<GUICollectionBox *>(m_ScenarioGUIController->GetControlUnderPoint(mouseX, mouseY, m_ScenarioCollectionBoxes.at(ScenarioCollections::PlayerSetupBox), 1));
+		if (hoveredCell) {
+			// Find which cell is being hovered over.
```

Comment:
This whole area is kind of a mess
1. It would be nice to have some local variables for defaultColour and hoveredColour so it's clearer from reading what colour setting is what
2. It seems odd to have this stuff separated from the stuff right below. I'd advocate for a total rewrite, but it doesn't really matter I guess :(
3. I'd at least like the comments here to be consolidated into one explaining what's generally being done, i.e. `This code determines which player and team are being hovered over, and also makes sure all non-hovered boxes are not highlighted.` Also, maybe the un-highlighting can go somewhere else?